### PR TITLE
add variables table

### PR DESCRIPTION
**Pull request by rpoluri**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,10 +3,32 @@
 
 For more information please refer to the main [Apiary](https://github.com/ExpediaInc/apiary) project page.
 
+## Variables
+|Environment Variable|Required|Description|
```
Comment (massdosage) [inline README.md]: For some of level of consistency across our documentation can we have the columns here match https://github.com/ExpediaInc/apiary-metastore/blob/master/VARIABLES.md (i.e. Name, Description, Default, Required)? Type is irrelevant for env vars I guess?

File: README.md [hunk 2]
```diff
@@ -3,10 +3,32 @@
 
 For more information please refer to the main [Apiary](https://github.com/ExpediaInc/apiary) project page.
 
+## Variables
```
Comment (massdosage) [inline README.md]: Variables -> Environment Variables

File: README.md [hunk 3]
```diff
@@ -3,10 +3,32 @@
...
+|Environment Variable|Required|Description|
+|----|----|----|
+|AWS_REGION|Yes|AWS region to configure various AWS clients.|
+|VAULT_ADDR|Yes|Address of Vault server for secrets.|
+|VAULT_LOGIN_PATH|No|Alternative login path on Vault server.|
+|vault_path|Yes|Vault path to read secrets from.|
```
Comment (massdosage) [inline README.md]: Shouldn't this be upper case?

File: README.md [hunk 4]
```diff
@@ -3,10 +3,32 @@
...
+|ENABLE_METRICS|No|Option to enable sending Hive Metastore metrics to CloudWatch.|
+|SNS_ARN|No|The SNS topic ARN to which metadata updates will be sent.|
```
Comment (massdosage) [inline README.md]: We probably should have called this METADATA_UPDATES_SNS_ARN, is that easy to change now?

File: README.md [hunk 5]
```diff
@@ -3,10 +3,32 @@
...
+|ENABLE_GLUESYNC|No|Option to turn on GlueSync Hive Metastore listener.|
+|GLUE_PREFIX|No|Prefix added to Glue databases to handle database name collisions when synchronizing multiple Hive Metastores to the Glue catalog.|
+|POLICY_MGR_URL|No|Ranger admin URL from where policies will be downloaded.|
```
Comment (massdosage) [inline README.md]: We probably should have called this RANGE_POLICY_MANAGER_URL or is this one that we have no control over as its part of Ranger? Again, not sure how easy this is to change now.

File: README.md [hunk 6]
```diff
@@ -3,10 +3,32 @@
...
+|RANGER_SERVICE_NAME|No|Ranger service name used to configure RangerAuth plugin.|
+|AUDIT_DB_URL|No|Ranger audit database JDBC URL.|
```
Comment (massdosage) [inline README.md]: RANGER_AUDIT_DB_URL

File: README.md [hunk 7]
```diff
@@ -3,10 +3,32 @@
...
+|LDAP_URL|No|Active Directory URL to enable group mapping in metastore.|
+|LDAP_BASE|No|LDAP base dn used to search for user groups.|
```
Comment (massdosage) [inline README.md]: dn -> DN

File: README.md [hunk 8]
```diff
@@ -3,10 +3,32 @@
...
+|instance_type|No|Hive Metastore instance type, applicable values are: readwrite, readonly|
```
Comment (massdosage) [inline README.md]: INSTANCE_TYPE?

Also, isn't this a bit confusing with AWS `instance` types? HIVE_METASTORE_ACCESS_MODE ? Something else?

File: README.md [hunk 9]
```diff
@@ -3,10 +3,32 @@
...
+|INSTANCE_NAME|Yes|Apiary instance name, will be used as prefix on most AWS resources to allow multiple Apiary instance deployments.|
+|HIVE_DBS|No|CSV list of hive schema names, when specified hive schemas will be created and mapped to corresponding S3 buckets.|
```
Comment (massdosage) [inline README.md]: hive -> Hive
CSV -> comma separated
schema names -> database names
HIVE_DBS -> HIVE_DB_NAMES ?

File: README.md [hunk 10]
```diff
@@ -3,10 +3,32 @@
...
+|EXTERNAL_DATABASE|No|Option to enable external database mode, when specified it disables managing Hive Metastore mysql database schema.|
```
Comment (massdosage) [inline README.md]: mysql -> MySQL

File: README.md [hunk 11]
```diff
@@ -3,10 +3,32 @@
...
+|loglevel|No|Hive Metastore service log4j log level.|
```
Comment (massdosage) [inline README.md]: HIVE_METASTORE_LOG_LEVEL ?

log4j -> Log4j

**Comment by rpoluri** [comment]
I agree we could name things better. most of these require changes in multiple repos docker,terraform& extensions.

If it is OK with everyone, I can go a head and make required changes. I think it better to change before 1.0 and keep these stable after 1.0.

**Comment by massdosage** [comment]
Yeah, the changes will touch quite a few repos but should be simple and straightforward. My vote goes for doing this now, will be harder to change later.

**Comment by mroark1m** [comment]
I agree, better to do it now.

File: README.md (hunk 1 above)
Comment (rpoluri) [reply README.md]: This is following same pattern as we are using in https://github.com/ExpediaInc/apiary-extensions/tree/master/apiary-metastore-listener

https://github.com/ExpediaInc/apiary-metastore/blob/master/VARIABLES.md has type and default which are not applicable here as type is alway string and default will be always blank.

**Comment by rpoluri** [comment]
Made most of changes requested. 
didn't update SNS_ARN as we need to publish new jar file to maven, will create new PR once that is done.

File: README.md [hunk 12]
```diff
@@ -3,10 +3,36 @@
...
+|AWS_REGION|Yes|AWS region to configure various AWS clients.|
+|ENABLE_METRICS|No|Option to enable sending Hive Metastore metrics to CloudWatch.|
+|ENABLE_GLUESYNC|No|Option to turn on GlueSync Hive Metastore listener.|
+|EXTERNAL_DATABASE|No|Option to enable external database mode, when specified it disables managing Hive Metastore MySQL database schema.|
+|GLUE_PREFIX|No|Prefix added to Glue databases to handle database name collisions when synchronizing multiple Hive Metastores to the Glue catalog.|
+|HADOOP_HEAPSIZE|No|Hive Metastore Java process heapsize.|
+|RANGER_POLICY_MANAGER_URL|No|Ranger admin URL from where policies will be downloaded.|
+|RANGER_SERVICE_NAME|No|Ranger service name used to configure RangerAuth plugin.|
+|RANGER_AUDIT_DB_URL|No|Ranger audit database JDBC URL.|
+|RANGER_AUDIT_SOLR_URL|No|Ranger SOLR audit URL.|
```
Comment (massdosage) [inline README.md]: SOLR -> Solr
(in the Description)

File: README.md [hunk 13]
```diff
@@ -3,10 +3,36 @@
...
+|LDAP_URL|No|Active Directory URL to enable group mapping in metastore.|
+|LDAP_BASE|No|LDAP base DN used to search for user groups.|
+|HIVE_METASTORE_ACCESS_MODE|No|Hive Metastore access mode, applicable values are: readwrite, readonly|
+|HIVE_DB_NAMES|No|comma separated list of hive database names, when specified hive databases will be created and mapped to corresponding S3 buckets.|
```
Comment (massdosage) [inline README.md]: hive -> Hive

File: files/ranger-hive-audit.xml [hunk 14]
```diff
@@ -52,17 +52,17 @@
 	<!--  Properties whose name begin with "xasecure.audit.jpa." are used to configure JPA -->
 	<property>
 		<name>xasecure.audit.jpa.javax.persistence.jdbc.url</name>
-		<value>AUDIT_DB_URL</value>
+		<value></value>
```
Comment (massdosage) [inline files/ranger-hive-audit.xml]: What are these changes about?

File: files/update_property.py [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+import os
+from xml.etree import ElementTree as ET
+from xml.parsers.expat import ExpatError
+
+def write_properties_to_xml(xml_path, property_name='', property_value=''):
+	if(os.path.isfile(xml_path)):
+		try:
+			xml = ET.parse(xml_path)
+		except ExpatError:
+			print "Error while parsing file:"+xml_path
```
Comment (massdosage) [inline files/update_property.py]: Put a space after the ":" so it's easier to read the file name

File: files/update_property.py [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+
+
+
+if __name__ == '__main__':
+	if(len(sys.argv) > 1):
+		if(len(sys.argv) > 3):
+			parameter_name = sys.argv[1] if len(sys.argv) > 1  else None
+			parameter_value = sys.argv[2] if len(sys.argv) > 2  else None
+			ranger_admin_site_xml_path = sys.argv[3] if len(sys.argv) > 3  else None
+		else:
```
Comment (massdosage) [inline files/update_property.py]: Can you turn this into an `elif` with the below line merged up here into one statement?

File: files/ranger-hive-audit.xml (hunk 14 above)
Comment (rpoluri) [reply files/ranger-hive-audit.xml]: previously was using sed to update XML properties, had to escape URLs to work with sed.
so while we are touching variables, fixed that. update_property.py works fine without escaping.

File: files/update_property.py (hunk 15 above)
Comment (rpoluri) [reply files/update_property.py]: This is a copy from ranger project, hence Apache license.
https://github.com/apache/ranger/blob/master/security-admin/scripts/update_property.py

File: files/ranger-hive-audit.xml (hunk 14 above)
Comment (massdosage) [reply files/ranger-hive-audit.xml]: Ah, OK.

File: files/update_property.py (hunk 15 above)
Comment (massdosage) [reply files/update_property.py]: OK, please add a comment at the top saying this.

File: files/startup.sh [hunk 17]
```diff
@@ -5,55 +5,57 @@
...
     update-ca-trust enable
 fi
 
 #configure ranger authorization
-if [[ -n $POLICY_MGR_URL ]]; then
+if [[ -n $RANGER_POLICY_MANAGER_URL ]]; then
     export METASTORE_PRELISTENERS="${METASTORE_PRELISTENERS},com.expedia.apiary.extensions.rangerauth.listener.ApiaryRangerAuthPreEventListener"
-    sed "s/POLICY_MGR_URL/${POLICY_MGR_URL}/" -i /etc/hive/conf/ranger-hive-security.xml
-    sed "s/RANGER_SERVICE_NAME/${RANGER_SERVICE_NAME}/" -i /etc/hive/conf/ranger-hive-security.xml
+    update_property.py ranger.plugin.hive.policy.rest.url ${RANGER_POLICY_MANAGER_URL} /etc/hive/conf/ranger-hive-security.xml
```
Comment (massdosage) [inline files/startup.sh]: Can you call the Python script directly like this? Don't you have to do `python update_property.py`? I'm guessing you've tried this out and it works, I'm just not sure how it works ;)

File: files/update_property.py (hunk 16 above)
Comment (massdosage) [reply files/update_property.py]: I guess if this is the code we've got from Ranger we just want to keep it like this?

File: files/startup.sh (hunk 17 above)
Comment (rpoluri) [reply files/startup.sh]: yes tested this.
added #! and set execute bit on the file.

File: files/update_property.py [hunk 18]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/python
```
Comment (massdosage) [inline files/update_property.py]: Ah right, this, OK.

**Pull request merged**